ravenml/data/write_dataset.py

```diff
@@ -0,0 +1,248 @@
+            source_dir=data_dir, dest_dir=self.temp_dir, condition_func=need_file)
+        
+        self.image_ids = image_ids
+        self.filter_metadata = filter_metadata
+    
+    @cli_spinner_wrapper("Writing out metadata locally...")
+    def write_metadata(self):
+        """Writes out a metadata file in JSON format
+
+        Args:
+            name (str): the name of the dataset
+            comments (str): comments or notes supplied by the user regarding the
+                dataset produced by this tool
+            training_type (str): the training type selected by the user
+            image_ids (list): a list of image IDs that ended up in the final
+                dataset (either dev or test)
+            filters (dict): a dictionary representing filter metadata
+            transforms (dict): a dictionary representing transform metadata
+            out_dir (Path, optional): Defaults to Path.cwd().
+        """
+        dataset_path = self.dataset_path / self.dataset_name
+        metadata_filepath = dataset_path / 'metadata.json'
+
+        metadata = {}
+        metadata["name"] = self.dataset_name
+        metadata["date_created"] = datetime.utcnow().isoformat() + "Z"
+        metadata["created_by"] = self.created_by
+        metadata["comments"] = self.comments
+        metadata["training_type"] = self.plugin_name
+        metadata["image_ids"] = self.image_ids
+        metadata["filters"] = self.filter_metadata
+        with open(metadata_filepath, 'w') as outfile:
+            json.dump(metadata, outfile) 
+
+    @cli_spinner_wrapper("Writing out dataset locally...")
+    def write_dataset(self, obj_list: list):
+        """Main driver for this file
+        
+        Args:
+            obj_list (list): objects that will be transformed into usable dataset
+            test_percent (float): percent of data that'll be test data
+            num_folds (int): number of folds to write
+            out_dir (Path): directory to write this dataset to
+            custom_dataset_name (str): name of the dataset's containing folder
+        """
+        dataset_path = self.dataset_path / self.dataset_name
+        print(dataset_path)
+        self.delete_dir(dataset_path) # Only deletes the contents already inside the directory
+
+        test_subset, dev_subset = self.split_data(obj_list)
+
+        # Test subset.
+        self.write_related_data(test_subset, dataset_path / 'test')
+
+        dev_path = dataset_path / 'splits'
+
+        # standard_path = dev_path / 'standard'
+        # write_out_fold(standard_path, fold, is_standard=True)
+
+        complete_path = dev_path / 'complete'
+        self.write_out_complete_set(complete_path, dev_subset)
+
+    def delete_dir(self, path):
+        """Deletes all files and folders in specified directory
+        
+        Args:
+            path (Path object): target directory
+        """
+        if os.path.isdir(path):
+            print('\n', "Deleting contents of", path,
```

Comment:
No reason to leave this in if we are using config files and prompting via `CreateInput`. users will have plenty of control over deletion, and will likely not be paying attention during this countdown anyway